common/test/acceptance/tests/test_cohorted_courseware.py

```diff
@@ -0,0 +1,226 @@
+from ..pages.lms.courseware import CoursewarePage
+from ..pages.lms.auto_auth import AutoAuthPage as LmsAutoAuthPage
+
+from bok_choy.promise import EmptyPromise
+
+
+@attr('shard_1')
+class EndToEndCohortedCoursewareTest(ContainerBase):
+
+    def setUp(self, is_staff=True):
+
+        super(EndToEndCohortedCoursewareTest, self).setUp(is_staff=is_staff)
+        self.staff_user = self.user
+
+        self.content_group_a = "Content Group A"
+        self.content_group_b = "Content Group B"
+
+        # Create a student who will be in "Cohort A"
+        self.cohort_a_student_username = "cohort_a_student"
+        self.cohort_a_student_email = "cohort_a_student@example.com"
+        StudioAutoAuthPage(
+            self.browser, username=self.cohort_a_student_username, email=self.cohort_a_student_email, no_login=True
+        ).visit()
+
+        # Create a student who will be in "Cohort B"
+        self.cohort_b_student_username = "cohort_b_student"
+        self.cohort_b_student_email = "cohort_b_student@example.com"
+        StudioAutoAuthPage(
+            self.browser, username=self.cohort_b_student_username, email=self.cohort_b_student_email, no_login=True
+        ).visit()
+
+        # Create a student who will end up in the default cohort group
+        self.cohort_default_student_username = "cohort default student"
+        self.cohort_default_student_email = "cohort_default_student@example.com"
+        StudioAutoAuthPage(
+            self.browser, username=self.cohort_default_student_username,
+            email=self.cohort_default_student_email, no_login=True
+        ).visit()
+
+        # Start logged in as the staff user.
+        StudioAutoAuthPage(
+            self.browser, username=self.staff_user["username"], email=self.staff_user["email"]
+        ).visit()
+
+    def populate_course_fixture(self, course_fixture):
+        """
+        Populate the children of the test course fixture.
+        """
+        self.group_a_problem = 'GROUP A CONTENT'
+        self.group_b_problem = 'GROUP B CONTENT'
+        self.group_a_and_b_problem = 'GROUP A AND B CONTENT'
+        self.visible_to_all_problem = 'VISIBLE TO ALL CONTENT'
+        course_fixture.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection').add_children(
+                    XBlockFixtureDesc('vertical', 'Test Unit').add_children(
+                        XBlockFixtureDesc('problem', self.group_a_problem, data='<problem></problem>'),
+                        XBlockFixtureDesc('problem', self.group_b_problem, data='<problem></problem>'),
+                        XBlockFixtureDesc('problem', self.group_a_and_b_problem, data='<problem></problem>'),
+                        XBlockFixtureDesc('problem', self.visible_to_all_problem, data='<problem></problem>')
+                    )
+                )
+            )
+        )
+
+    def _enable_cohorts_in_course(self):
```

Comment:
I'm not sure when it became a pattern to prefix helper methods in test classes with `_`, but I know I'm guilty of contributing. I'm just wondering why we do it - seems unnecessary to me, since the only methods that are run as tests are those beginning with `test`.